app/controllers/api/v1/participant_declarations_controller.rb

```diff
@@ -3,19 +3,37 @@
 module Api
   module V1
     class ParticipantDeclarationsController < Api::ApiController
+      include ActionController::StrongParameters
       include ApiTokenAuthenticatable
-      before_action :set_paper_trail_whodunnit
+      wrap_parameters format: [:json]
 
       def create
-        return head :not_found unless params[:id]
+        config = HashWithIndifferentAccess.new({ raw_event: request.raw_post, lead_provider: current_user }).merge(declaration_params)
+        return head(RecordParticipantEvent.call(config)) if check_config(config)
 
-        user = User.find(params[:id])
+        raise ActionController::ParameterMissing, missing_params(config)
+      end
+
+    private
 
-        # TODO: Switch on declaration type
-        # TODO: Confirm that this participant is relevant to this lead provider
-        return head :not_modified unless InductParticipant.call({ lead_provider: current_user, early_career_teacher_profile: user.early_career_teacher_profile })
+      def permitted_params
+        RecordParticipantEvent.required_params
+      end
+
+      def required_params
+        %w[participant_id declaration_date declaration_type]
+      end
+
+      def missing_params(config)
+        required_params - config.keys
+      end
+
+      def check_config(config)
+        missing_params(config).empty?
+      end
 
-        head :no_content
+      def declaration_params
+        params.permit(:participant_id, :declaration_date, :declaration_type)
```

Comment:
Did you intend to use `permitted_params` and `required_params` here?

Reply:
Enabling StrongParameters didn't work as expected, and the incoming API wasn't wrapping in such a way that `required_params` could be used. I may be missing something from that, so if you can enlighten....